manim/utils/decorators.py

```diff
@@ -0,0 +1,454 @@
+"""Function decorators."""
+
+__all__ = ["deprecated", "deprecated_params"]
+
+
+import re
+from typing import Any, Callable, Iterable, List, Optional, Tuple, Union
+
+from decorator import decorate, decorator
+
+from .. import logger
+
+
+def __get_callable_info(callable: Callable) -> Tuple[str, str]:
+    """Returns type and name of a callable.
+
+    Parameters
+    ----------
+    callable
+        The callable
+
+    Returns
+    -------
+    Tuple[str, str]
+        The type and name of the callable. Type can can be one of "class", "method" (for
+        functions defined in classes) or "function"). For methods, name is Class.method.
+    """
+    what = type(callable).__name__
+    name = callable.__qualname__
+    if what == "function" and "." in name:
+        what = "method"
+    elif what == "type":
+        what = "class"
+    return (what, name)
+
+
+def __deprecation_text_component(
```

Comment:
From [PEP 8](https://www.python.org/dev/peps/pep-0008/), it's good convention to use single underscore (`_get_callable_info`, `_deprecation_text_component`) to indicate methods of internal use. Double underscores are often used to define private variables/methods within classes, and would trigger name mangling (class Foo: `__bar` becomes `_Foo__bar`).